docs/designs/013-audit-logging.md

```diff
@@ -0,0 +1,735 @@
+
+#### Built-in Sink Implementations
+
+```go
+// DatastoreSink writes audit logs to the NVSentinel datastore
+type DatastoreSink struct {
+    store datastore.AuditStore  // Uses existing datastore interface
+}
+
+func (s *DatastoreSink) Name() string { return "datastore" }
+
+func (s *DatastoreSink) Write(ctx context.Context, log *AuditLog) error {
+    return s.store.InsertAuditLog(ctx, log)
+}
+
+// K8sEventSink creates Kubernetes events for audit logs
+type K8sEventSink struct {
+    client    kubernetes.Interface
+    namespace string
+}
+
+func (s *K8sEventSink) Name() string { return "k8s-events" }
+
+func (s *K8sEventSink) Write(ctx context.Context, log *AuditLog) error {
+    return s.createEvent(ctx, log)
+}
+
+// Future: CloudLoggingSink for GCP/AWS/Azure logging
+// type CloudLoggingSink struct { ... }
+```
+
+### Automatic Capture via HTTP RoundTripper
+
+Instead of manually adding audit calls everywhere, we wrap the client-go HTTP transport:
+
+```go
+// AuditingRoundTripper wraps http.RoundTripper to automatically capture write operations
+type AuditingRoundTripper struct {
+    delegate  http.RoundTripper
+    logger    *Logger
+    component ComponentType
+}
+
+func NewAuditingRoundTripper(delegate http.RoundTripper, logger *Logger, component ComponentType) *AuditingRoundTripper {
+    return &AuditingRoundTripper{
+        delegate:  delegate,
+        logger:    logger,
+        component: component,
+    }
+}
+
+func (rt *AuditingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+    // Execute the actual request
+    resp, err := rt.delegate.RoundTrip(req)
+    
+    // Only audit write operations (POST, PUT, PATCH, DELETE)
+    if isWriteMethod(req.Method) {
+        rt.auditRequest(req, resp)
+    }
+    
+    return resp, err
+}
+
+func (rt *AuditingRoundTripper) auditRequest(req *http.Request, resp *http.Response) {
+    // Parse K8s API path to extract resource info
+    // e.g., /api/v1/nodes/gpu-node-01 -> ResourceType: Node, ResourceName: gpu-node-01
+    resourceInfo := parseK8sAPIPath(req.URL.Path)
+    
+    rt.logger.Log(&AuditLog{
+        Component:         rt.component,
+        Operation:         httpMethodToOperation(req.Method),
+        ResourceType:      resourceInfo.Type,
+        ResourceName:      resourceInfo.Name,
+        ResourceNamespace: resourceInfo.Namespace,
+        NodeName:          extractNodeName(resourceInfo),
+        HealthEventID:     extractHealthEventID(resourceInfo),  // Extract from CR name
+        Description:       fmt.Sprintf("%s %s", req.Method, req.URL.Path),
+    })
+}
+
+func isWriteMethod(method string) bool {
+    return method == "POST" || method == "PUT" || method == "PATCH" || method == "DELETE"
+}
+
+// extractHealthEventID extracts the health event ID from maintenance CR names
+// CR name format: maintenance-<nodeName>-<healthEventID>
+// Example: maintenance-gpu-node-01-abc123def456
+func extractHealthEventID(resourceInfo ResourceInfo) string {
```

Comment:
I'm not a fan of this. We can change the CR format in future and this will break logging. We should try and see if there is a more structured approach if we really need this. I suspect we won't